src/constants.md

```diff
@@ -1,10 +1,22 @@
 #### Constants
 
-Constants are one of the two things that can be defined in both the module scope and a function scope (the other being more functions). The syntax is simply:
+Module-level constants allow the user to define fixed data that is shared across functions (or exported to be shared across modules). The syntax is simply:
 
-```
+```rust
 const varname: typename = constvalue
+const othervar = otherconstval
 ```
 
-where typename must be an existing type and the `constvalue` must be replaced with a number, boolean, string, etc. (Actually no etc, yet, as user-defined types can't be defined as a constant, yet, but that will change!)
+The `typename` is not required due to type inference, but can be added for clarity.
+
+For those coming from Javascript/Typescript, `const` in `alan` is much more strict. A constant is truely constant, not just a variable that can't be reassigned with a new struct. This means the following is invalid:
+
+```rust
+const pushedVals = new Array<int64> [ ]
+
+fn saveForLater(val: int64) {
+  pushedVals.push(val)
+}
+```
 
+Shared mutable global state inherently prevents parallelization, so it is discouraged, but provided through [a standard library](./std_datastore.md) to allow safe operations on shared, mutable state with explicit guarantees (or not, if not needed) on locking and read-after-write.
```

Comment:
remove the `std_datastore` reference?

Reply:
ah, yes. That does remind me, though, that people will have a hard time writing non-trivial applications without `datastore`. I think we should make an RFC for it before launch.

Reply:
Any suggestions on how to adjust this paragraph? I'm kinda stumped.

Reply:
> ah, yes. That does remind me, though, that people will have a hard time writing non-trivial applications without `datastore`. I think we should make an RFC for it before launch.

I don't think it blocks launch since we will have an RFC for a postgres client, but I do agree there should be an RFC at some point.

> Any suggestions on how to adjust this paragraph? I'm kinda stumped.

I would just remove it for now?

Reply:
:+1: